app/lib/content_security_policy.rb

```diff
@@ -13,8 +13,44 @@ def media_host
     cdn_host_value || assets_host
   end
 
+  def sso_host
+    @sso_host ||= sso_host_from_config
+  end
+
   private
 
+  def sso_host_from_config
+    return unless valid_sso_host_config?
+
+    provider = Devise.omniauth_configs[Devise.omniauth_providers[0]]
+    begin
+      case provider.provider
+      when :cas
+        provider.cas_url
+      when :saml
+        provider.options[:idp_sso_target_url]
+      when :openid_connect
+        provider.options.dig(:client_options, :authorization_endpoint) || OpenIDConnect::Discovery::Provider::Config.discover!(provider.options[:issuer]).authorization_endpoint
+      end
+    end
+  end
+
+  def valid_sso_host_config?
```

Comment:
There are a few other spots in codebase which do essentially this same check. Might make sense at some point to pull out a `SingleSignOnConfiguration` (or whatever) class to roll up that logic in one place.